crates/core/src/factories/core_space.rs

```diff
@@ -198,6 +198,39 @@ impl TxSpaceHandler for TxHandlerTranslator {
     ) -> K2Result<()> {
         self.0.recv_notify(peer, space, data)
     }
+
+    fn mark_peer_unresponsive(&self, peer: Url) -> K2Result<()> {
+        let core_space = self.1.upgrade().ok_or(K2Error::Other {
+            ctx: "CoreSpace had been dropped.".into(),
+            src: DynInnerError(None),
+        })?;
+        tokio::task::spawn(async move {
+            // Only add a peer as unreachable to the peer meta store if it is
+            // also in the peer store. That's because this method may be called
```

Comment:
Is that necessary to assume? yes the transport doesn't know, but it's looking up the right space based on the space id. So while this is technically possible, I think it's okay to accept the input

Reply:
We just talked about this and I was mistaken. We identified three ways of doing this
1. the way it's implemented on this PR
2. mark URLs unresponsive in all spaces' peer stores (might become a lot?)
3. add peer stores to the transport, so that it can search through them to find where the peer with the URL is

I think how it's done on this PR is fine.

Reply:
I see, because the space isn't available in the transport at this point. Should that be changed though? This `send` is only called from `send_space_notify` and `send_module`, which doesn't pass the space down because it's in the encoded message. You could decode the message on this code path but it'd probably be cheaper to just pass the space id. Then you do know which space to target and don't have to broadcast this?

Reply:
Aah, yes, that's best, I think.

Reply:
> Then you do know which space to target and don't have to broadcast this?

I actually think we'd want to broadcast it anyway because the peer url likely doesn't only belong to a single space and we'd want to mark that url as unreachable for all spaces that it's part of I assume. At least that was my thinking.

Reply:
If we think it makes sense to forward the space id anyway since we're going to break the API with the next release I'm fine with that. But I think we still want to iterate through all spaces. I should however probably update the doc comment then because wanting to mark a url as unreachable for all spaces that it's part of is the primary reason then and not the lack of knowledge of the space id.